packages/alfa-rules/src/sia-r73/rule.ts

```diff
@@ -0,0 +1,76 @@
+import { Rule } from "@siteimprove/alfa-act";
+import { Element, Namespace } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Ok, Err } from "@siteimprove/alfa-result";
+import { Style } from "@siteimprove/alfa-style";
+import { Page } from "@siteimprove/alfa-web";
+
+import { expectation } from "../common/expectation";
+
+import { isVisible } from "../common/predicate/is-visible";
+
+const { isElement, hasName, hasNamespace } = Element;
+const { and } = Predicate;
+
+export default Rule.Atomic.of<Page, Element>({
+  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r73.html",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return document
+          .descendants({
+            flattened: true,
+            nested: true,
+          })
+          .filter(
+            and(
+              isElement,
+              and(hasNamespace(Namespace.HTML), hasName("p"), isVisible(device))
+            )
+          );
+      },
+
+      expectations(target) {
+        const style = Style.from(target, device);
+
+        const { value: lineHeight } = style.computed("line-height");
+        const { value: fontSize } = style.computed("font-size");
+
+        return {
+          1: expectation(
+            lineHeight.type === "number" && lineHeight.value >= 1.5,
+            () => Outcomes.IsSufficient,
+            () =>
+              expectation(
+                lineHeight.type === "length" &&
+                  lineHeight.value / fontSize.value >= 1.5,
+                () => Outcomes.IsSufficient,
+                () =>
+                  expectation(
+                    lineHeight.type === "keyword" &&
+                      lineHeight.value === "normal",
+                    () => Outcomes.IsNormal,
+                    () => Outcomes.IsInsufficient
+                  )
+              )
+          ),
+        };
+      },
+    };
```

Comment:
Wondering if this would read better with a `switch (lineHeight.type)` instead. Notably for not going through everything when we pass the `type` test but fail the `value` one.
```suggestion
switch (lineHeight.type) {
  case "number":
    return expectation(
      lineHeight.value >= 1.5,
      () => Outcomes.IsSufficient,
      () => Outcomes.IsInsufficient
      );
    case "length":
      …
    defaut: return Outcomes.IsInsufficient;
```